src/run.js

```diff
@@ -25,10 +25,11 @@ async function getGhCli(version) {
 }
 
 async function downloadGhCli(version) {
-  const toolDirectoryName = `gh_${version}_linux_amd64`;
-  const downloadUrl = `https://github.com/cli/cli/releases/download/v${version}/gh_${version}_linux_amd64.tar.gz`;
-  console.log(`downloading ${downloadUrl}`);
 
+  let architecture = process.arch
```

Comment:
I'm afraid this will not work. `process.arch` returns x64 when we need amd64. you can give it another shot using below pseudocode

```
let architecture = "amd64"
if process.arch in ["arm", "arm64"] {
  architecture = process.arch
}
```

can you please test the changes using at least one amd64 and one arm64 machine?